dagger-functions/src/test/java/io/odpf/dagger/functions/udfs/python/PythonUdfManagerTest.java

```diff
@@ -0,0 +1,178 @@
+        pythonUdfManager.registerPythonFunctions();
+
+        verify(configuration, times(1)).setString("python.files", pathFile);
+        verify(configuration, times(1)).setString("python.archives", "/path/to/file.txt");
+        verify(configuration, times(1)).setString("python.requirements", "requirements.txt");
+        verify(configuration, times(1)).setInteger("python.fn-execution.arrow.batch.size", 10000);
+        verify(configuration, times(1)).setInteger("python.fn-execution.bundle.size", 100000);
+        verify(configuration, times(1)).setLong("python.fn-execution.bundle.time", 1000);
+        verify(tableEnvironment, times(1)).executeSql(sqlRegisterFirstUdf);
+        verify(tableEnvironment, times(1)).executeSql(sqlRegisterSecondUdf);
+    }
+
+    @Test
+    public void shouldNotRegisterConfigIfNotSet() throws IOException {
+        String pathFile = getPath("python_udf.zip");
+
+        when(tableEnvironment.getConfig()).thenReturn(tableConfig);
+        when(tableConfig.getConfiguration()).thenReturn(configuration);
+        when(pythonUdfConfig.getPythonFiles()).thenReturn(pathFile);
+
+        PythonUdfManager pythonUdfManager = new PythonUdfManager(tableEnvironment, pythonUdfConfig);
+        pythonUdfManager.registerPythonFunctions();
+
+        verify(configuration, times(1)).setString("python.files", pathFile);
+        verify(configuration, times(0)).setString("python.archives", "/path/to/file.txt");
+        verify(configuration, times(0)).setString("python.requirements", "requirements.txt");
+    }
+
+    @Test
+    public void shouldRegisterPythonUdfFromPyFile() throws IOException {
+        String pathFile = getPath("test_udf.py");
+        String sqlRegisterUdf = "CREATE TEMPORARY FUNCTION TEST_UDF AS 'test_udf.test_udf' LANGUAGE PYTHON";
+
+        when(tableEnvironment.getConfig()).thenReturn(tableConfig);
+        when(tableConfig.getConfiguration()).thenReturn(configuration);
+        when(pythonUdfConfig.getPythonFiles()).thenReturn(pathFile);
+
+        PythonUdfManager pythonUdfManager = new PythonUdfManager(tableEnvironment, pythonUdfConfig);
+        pythonUdfManager.registerPythonFunctions();
+
+        verify(configuration, times(1)).setString("python.files", pathFile);
+        verify(tableEnvironment, times(1)).executeSql(sqlRegisterUdf);
+    }
+
+    @Test
+    public void shouldOnlyExecutePyFormatInsideZipFile() throws IOException {
+        String pathFile = getPath("python_udf.zip");
+
+        when(tableEnvironment.getConfig()).thenReturn(tableConfig);
+        when(tableConfig.getConfiguration()).thenReturn(configuration);
+        when(pythonUdfConfig.getPythonFiles()).thenReturn(pathFile);
+
+        PythonUdfManager pythonUdfManager = new PythonUdfManager(tableEnvironment, pythonUdfConfig);
+        pythonUdfManager.registerPythonFunctions();
+
+        verify(configuration, times(1)).setString("python.files", pathFile);
+        verify(tableEnvironment, times(2)).executeSql(anyString());
```

Comment:
can we check for times(0) for txt file?

Reply:
input for metode executeSql is string sql template for registering python function. For txt file it didn't generate that.

Should I manually create string sql template for txt file like it is a py file and test that for time(0)?

Reply:
ya then, instead of anyString, verify explicitly for sql queries